Add new material

Czino opened the pull request:
Please review these new submissions.

There's also a minor fix to avoid having the bot answer on its own tweets (in future we shall keep features and content in separate branches)

pox commented on material/economics/bitcoin-is-not-money.toml:
```diff
@@ -1,4 +1,4 @@
-tags="money"
+tags="money, bitcoinstandard"
```
It might be better to use an inline array `["money", "bitcoinstandard"]` instead of a comma-separated string. This way the TOML parser already converts it to an array.

Czino replied:
I had the same thought but decided to use a simple string instead to ease contribution, exceptions are links as an array broken in multiple lines is more readable.

pox commented on run.mjs:
```diff
@@ -52,7 +52,7 @@ function run() {
           message = [
             'Beep boop, I am @btcfudbuster 🤖',
-            'Mention me like this: "@btcfudbuster help" and a topic hashtag that I know, I will reply with a relevant article.',
+            'Mention me like this: "@btcfudbuster debunk" and a topic hashtag that I know, I will reply with a relevant article.',
```
how about `@btcfudbuster answer ...`?

pox reviewed (approved):
Looks good. Added some comments.
General notes:
1. The answers you're drafted are really good IMO. Nice work.
2. It's better to separate unrelated changes into separate PRs. e.g. the syntax changes have nothing to do with the curation.

pox commented:
I think it's a good idea to make the `getMostRelevantAnswer` function deterministic, for the sake of testability. Otherwise it will be a big PITA to test it (unit tests and randomness don't really go hand in hand).

Also, let's please merge and close this PR soon, instead of adding more and more commits to it. It makes it harder to review. (Branches should be short-lived, in general, is my thinking).

I see you still consider non-hashtag keywords (with lower weight). It's not a huge amount of code, so I don't think it's a huge problem, but I want to stress that I think it's not KISS. I don't think we stand to gain much from it, and it add complexity, and complexity has a cost in terms of maintaining the codebase, and also in terms of teaching people how the bot works predictably.

pox commented on answerUtils.mjs:
```diff
@@ -7,6 +7,10 @@ const __dirname = path.resolve()
 const allMaterial = glob.sync(__dirname + '/material/**/*.toml')
```
I recommend using `path.join` for combining path parts. More portable.

Czino replied:
changed

Czino commented:
Let us know how you'd structure `getMostRelevantAnswer`. I added one unit test that expects *a* result or none for garbage. I refrain from asserting the function to return a specific answer as they might be prone to change and it's impossible to test all cases as there are nearly infinite.

Fully agree on closing this branch, just waiting for BitcoinMemeHub to do the review on the material so we can move on.

I removed the checks for non-hashtag tokens. This way the bot can give 2 answers based on the tweet/results for more clarity (not implemented yet):

- no hashtags at all ("beep boop pls use hashtags to choose topics")
- no results ("beep boop I don't know about this fud yet")

pox commented:
The point of a unit test of `getMostRelevantAnswer` should be to validate the correctness of the algorithm against a known test fixture. If the algorithm essentially grades answers by relevance, then a unit test should assert the correctness of the grading. No randomness involved.
Then to choose with some randomness you can use some package like `random-weighted-choice`
Hope this is clearer now.

Czino commented:
I will open a separate issue for unit testing this method the way you described
